packages/instantsearch-react/src/themeable.js

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import reactThemeable from 'react-themeable';
+
+export default function themeable(defaultTheme) {
+  return Composed => {
+    function Themeable(props) {
+      const {theme = defaultTheme, ...otherProps} = props;
+      const applyTheme = reactThemeable(theme);
+
+      return <Composed {...otherProps} applyTheme={applyTheme} />;
+    }
+
+    Themeable.propTypes = {
+      theme(props, propName, componentName) {
+        const prop = props[propName];
```

Comment:
What about moving that prop type checker to another file? Or at least another function to be able to keep the same proptypes structure?